### PR TITLE
Split Coverity scan into a separate test

### PR DESCRIPTION
**Pull request by sgallagher**
Allows us to more easily exclude it for pull-requests and to speed
up the Fedora tests slightly.

Signed-off-by: Stephen Gallagher <sgallagh@redhat.com>

### CODE REVIEW

File: .travis/travis-coverity.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+sed -e "s/@IMAGE@/$repository\/$os:$release/" \
+    $SCRIPT_DIR/fedora/Dockerfile.deps.tmpl > $SCRIPT_DIR/fedora/Dockerfile.deps.$release
+
+sudo docker build -f $SCRIPT_DIR/fedora/Dockerfile.deps.$release -t fedora-modularity/libmodulemd-deps-$release .
+sudo docker build -f $SCRIPT_DIR/coverity/Dockerfile -t fedora-modularity/libmodulemd-coverity --build-arg TARBALL=$TARBALL .
+
+rm -f $TARBALL_PATH $SCRIPT_DIR/fedora/Dockerfile.deps.$release $SCRIPT_DIR/fedora/Dockerfile-$release
+
+# Override the standard tasks with the doc-generation
```
Comment (mmathesius) [inline .travis/travis-coverity.sh]: doc-generation? (A leftover from `travis-docs.sh`?)

File: .travis/travis-coverity.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+docker run \
+    -e COVERITY_SCAN_TOKEN=$COVERITY_SCAN_TOKEN \
+    -e TRAVIS=$TRAVIS \
+    -e TRAVIS_JOB_NAME="$TRAVIS_JOB_NAME" \
+    -e TRAVIS_COMMIT="$TRAVIS_COMMIT" \
+    -e DOC_TOKEN="$DOC_TOKEN" \
```
Comment (mmathesius) [inline .travis/travis-coverity.sh]: Is `DOC_TOKEN` needed here?

File: .travis/travis-coverity.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    $SCRIPT_DIR/fedora/Dockerfile.deps.tmpl > $SCRIPT_DIR/fedora/Dockerfile.deps.$release
+
+sudo docker build -f $SCRIPT_DIR/fedora/Dockerfile.deps.$release -t fedora-modularity/libmodulemd-deps-$release .
+sudo docker build -f $SCRIPT_DIR/coverity/Dockerfile -t fedora-modularity/libmodulemd-coverity --build-arg TARBALL=$TARBALL .
+
+rm -f $TARBALL_PATH $SCRIPT_DIR/fedora/Dockerfile.deps.$release $SCRIPT_DIR/fedora/Dockerfile-$release
+
+# Override the standard tasks with the doc-generation
+docker run \
+    -e COVERITY_SCAN_TOKEN=$COVERITY_SCAN_TOKEN \
```
Comment (mmathesius) [inline .travis/travis-coverity.sh]: I understand `COVERITY_SCAN_TOKEN` being needed here, but is it (still) needed in `travis-archlinux.sh`, `travis-centos.sh`, `travis-docs.sh`, `travis-fedora.sh`, `travis-mageia.sh`, and `travis-opensuse.sh`?

Comment (sgallagher) [reply .travis/travis-coverity.sh]: There's a lot of unnecessary copy-paste going on there, yeah. I'll fix that up.

File: .travis/travis-coverity.sh (hunk 2 above)
Comment (sgallagher) [reply .travis/travis-coverity.sh]: Nope, copy-paste error.

**Comment by sgallagher** [comment]
@mmathesius I updated the patch and also added a follow-up patch to reflow the `docker build` and `docker run` commands in the CI to make them more readable for future maintainers.

**Pull request merged**